src/components/ebay-carousel/index.js

```diff
@@ -85,120 +85,104 @@ function init() {
 
     this.subscribeTo(resizeUtil).on('resize', refresh.bind(this));
     this.refresh();
-    this.refresh(); // FIXME: currently needs a second call in v4
+}
+
+function onUpdate() {
+    this.processIndexChange();
+}
+
+function onDestroy() {
+    cancelAnimationFrame(this.processMovementFrame);
 }
 
 function refresh() {
     this.calculateWidths(true);
     if (this.state.isDiscrete) {
         this.simulateDotClick(this.state.slide);
-    } else if (this.state.isContinuous) {
-        this.performSlide();
     }
+    this.processMovement();
 }
 
-function update_index() { // eslint-disable-line camelcase
-    this.performSlide();
-}
-
-function handleNext(originalEvent) {
-    if (!this.state.nextControlDisabled) {
-        if (this.state.isDiscrete) {
-            this.simulateDotClick(this.state.slide + 1);
-        } else if (this.state.isContinuous) {
-            this.setState('index', this.calculateNextIndex());
-        }
-        emitAndFire(this, 'carousel-next', { originalEvent });
+/**
+ * High level function called for movement upon changing index (via UI and API)
+ * Exits early if there was no actual change to index
+ */
+function processIndexChange() {
+    // TODO: API manipulation is disabled in native scroll case
+    if (this.usesNativeScroll) {
+        return;
     }
-}
 
-function handlePrev(originalEvent) {
-    if (!this.state.prevControlDisabled) {
-        if (this.state.isDiscrete) {
-            this.simulateDotClick(this.state.slide - 1);
-        } else if (this.state.isContinuous) {
-            this.setState('index', this.calculatePrevIndex());
-        }
-        emitAndFire(this, 'carousel-prev', { originalEvent });
+    if (this.state.index > this.lastIndex) {
```

Comment:
Shouldn't this be `this.state.index !== this.lastIndex`? I might be missing something.

Reply:
`lastIndex` is used as in "A carousel of length 4 would have lastIndex=3." I'm using `activeIndex` below to keep track of when to handle the state updates.